### ISSUE TITLE
DeliversCash: and AcceptsDeliveredCash: related issues.

### ISSUE THREAD

**Issue by MlemandPurrs** [OpenRA/OpenRA#18087]
In my RA-based mod have decided to make production structures able to recycle units for half(or quarter) their value, in addition to producing them. This was done via adding an DeliversCash: to every producible unit with differing Type given, that is one different per production queue, and a AcceptsDeliveredCash: with the corresponding Type on every production building, with the exception being Aircrafts get recycled on the Service Depot instead (as previously had reallocated repair functions for them to Airfield/Helipad) 
The end result is that while it works, as there is no way to tell AcceptsDeliveredCash: to
accept only the player's own units, they can be ordered to enter an allied Production building too, unintentionally making every unit into an armed(or flying in the case of aircrafts) supply truck (of sorts). Furthermore, since there is no way to tell DeliversCash to use an different cursor, it can be very confusing for the player in the case of Aircraft, as upon seeing the enter cursor might order it to land on the Service Depot thinking it will be repaired only to realize it will get sold instead.
Also, while setting this up have realized how better it would be if the DeliversCash: simply defaults to the unit's cost value, rather than being fixed at 500, requiring manually to be changed on each unit.
edit: Tried the Aircraft landing on Service Depot part. It seems while the Aircraft does indeed land on the Service Depot, unit is not getting sold. Might have something to do with all its footprint being passable cell types, unsure.

TL;DR:
there should be a way to tell AcceptsDeliveredCash: to accept only your own units.
DeliversCash: should have support for defining a cursor.
DeliversCash: should probably default to the unit's value instead.

**Comment by ABrandau** [comment]
I would say it should have an option to set the value based off a percentage of the cost of the unit, or a fixed amount.
